detect_secrets/util.py

```diff
@@ -1,8 +1,52 @@
+import hashlib
 import os
 import subprocess
+import sys
 
 
-def get_root_directory():       # pragma: no cover
+def is_python_2():
+    return sys.version_info[0] < 3
+
+
+def build_automaton(word_list):
+    """
+    :type word_list: str
+    :param word_list: optional word list file for ignoring certain words.
+
+    :rtype: (ahocorasick.Automaton, str)
+    :returns: an automaton, and an iterated sha1 hash of the words in the word list.
+    """
+    # Dynamic import due to optional-dependency
+    try:
+        import ahocorasick
+    except ImportError:  # pragma: no cover
+        print('Please install the `pyahocorasick` package to use --word-list')
+        raise
+
+    # See https://pyahocorasick.readthedocs.io/en/latest/
+    # for more information.
+    automaton = ahocorasick.Automaton()
+    word_list_hash = ''
+
+    with open(word_list) as f:
+        for line in f:
+            # .lower() to make everything case-insensitive
+            line = line.lower().strip()
+            if len(line) > 3:
+                word_list_hash = hashlib.sha1(
```

Comment:
Is there a reason we want to include the previous hash? It seems like we're not trying to keep this actually secure.

What about making 1 hash object at the beginning and then updating it whenever we add a new word? I think this saves on the number of hashing operations.

e.g. 
```
hash = hashlib.sha1()
for bla in blah:
    hash.update(bla)
...
word_hash_list = hash.hexdigest()
return (automaton, word_hash_list)
```

Reply:
I totally forgot about the `.update` function, that's a good point, as either way it'll change when a word is added.

Reply:
👍